src/app/shared/components/addscenario/addscenario.component.ts

```diff
@@ -152,6 +162,82 @@ export class AddScenarioModal implements OnInit, OnDestroy {
                 this.cancelCreateScenario();
             }
         );
+
+        if (this.clone == true){
+            this.clearScenario();
+            this.cloneSenario();
+        }  
+    }
+
+    cloneSenario(){  
+        this.unitTypes.forEach( (element,index) => {  
+            if (element.id.toString() == this.itemParameters.unit.id.toString()){
+                this.newScenForm.patchValue({regressionRegions: {regressions:{unit:this.unitTypes[index]}}});
+            }
+        });
+
+        this.newScenForm.patchValue({
+            statisticGroupID: this.statisticGroup.toString(),
+            regressionRegions: {
+                ID: this.regRegion.id.toString(),
+                regressions:{
+                    ID: this.itemParameters.id.toString(),
+                    equation: this.itemParameters.equation.toString(),
+                    equivalentYears: this.itemParameters.equivalentYears.toString(),
```

Comment:
I had issues with this when cloning one of the regressions from the test region. I think this is because I didn't get it a value for equivalent years when I created it. Instead of it being 0, it's coming back as undefined so it can't be converted using `.toString()` here, causing a lot of the cloning to fail.

I also noticed that when I clone regressions that do have values for equivalent years it gets reset to 0 when cloned. I didn't dive into it too much, but I noticed this occurred even if I remove the `.toString()` method. It might be getting overwritten at some point.